File: src/main/java/dev/himirai/sampleplugin/SamplePlugin.java
package dev.himirai.sampleplugin;

import lombok.Getter;
import org.bukkit.Bukkit;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.AsyncPlayerPreLoginEvent;
import org.bukkit.event.player.PlayerJoinEvent;
import org.bukkit.event.player.PlayerQuitEvent;
import org.bukkit.plugin.java.JavaPlugin;

import java.util.logging.Level;

public class SamplePlugin extends JavaPlugin implements Listener {

	@Getter
	private static SamplePlugin plugin;

	@Override
	public void onEnable() {
		plugin = this;
		Bukkit.getPluginManager().registerEvents(this, this);
	}

	@EventHandler
	public void on(AsyncPlayerPreLoginEvent event) {
		getLogger().log(Level.INFO, "Player " + event.getPlayerProfile().getName() + " pre-logging in...");
	}

	@EventHandler
	public void on(PlayerJoinEvent event) {
		event.joinMessage(null);
		getLogger().log(Level.INFO, "Player " + event.getPlayer().getName() + " connected!");
	}

	@EventHandler
	public void on(PlayerQuitEvent event) {
		event.quitMessage(null);
		getLogger().log(Level.INFO, "Player " + event.getPlayer().getName() + " disconnected");
	}

}

File: src/main/kotlin/dev/himirai/sampleplugin/SamplePlugin.kt
package dev.himirai.sampleplugin

import org.bukkit.Bukkit
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.AsyncPlayerPreLoginEvent
import org.bukkit.event.player.PlayerJoinEvent
import org.bukkit.event.player.PlayerQuitEvent
import org.bukkit.plugin.PluginDescriptionFile
import org.bukkit.plugin.java.JavaPlugin
import org.bukkit.plugin.java.JavaPluginLoader
import java.io.File
import java.util.logging.Level

lateinit var PLUGIN: SamplePlugin

open class SamplePlugin : JavaPlugin, Listener {

	constructor() : super()

	constructor(loader: JavaPluginLoader, description: PluginDescriptionFile, dataFolder: File, file: File) : super(loader, description, dataFolder, file)

	override fun onEnable() {
		PLUGIN = this
		Bukkit.getPluginManager().registerEvents(this, this)
	}

	@EventHandler
	fun AsyncPlayerPreLoginEvent.on() {
		logger.log(Level.INFO, "Player ${playerProfile.name} pre-logging in...")
	}

	@EventHandler
	fun PlayerJoinEvent.on() {
		logger.log(Level.INFO, "Player ${player.name} connected!")
		player.level = 100
	}

	@EventHandler
	fun PlayerQuitEvent.on() {
		logger.log(Level.INFO, "Player ${player.name} disconnected")
		player.level = 0
	}

}
